src/rust/src/pkcs7.rs

```diff
@@ -407,6 +566,39 @@ fn smime_canonicalize(data: &[u8], text_mode: bool) -> (Cow<'_, [u8]>, Cow<'_, [
     }
 }
 
+fn smime_decanonicalize(data: &[u8], text_mode: bool) -> Cow<'_, [u8]> {
```

Comment:
@reaperhulk can I bug you to review this, I feel like I could use a second pair of eyes.